bin/Debugger.Sample/Debugger.Sample.cpp

```diff
@@ -260,6 +267,65 @@ JsErrorCode DefineHostCallback(
     return JsNoError;
 }
 
+JsErrorCode RedirectConsoleToDebugger(DebugProtocolHandler *handler)
+{
+    // Get global.console object. If that is not defined just return, otherwise get the debugger.console and patch with global.console object
+
+    JsValueRef globalObject = JS_INVALID_REFERENCE;
+    IfFailRet(JsGetGlobalObject(&globalObject));
+
+    JsPropertyIdRef consolePropertyId = JS_INVALID_REFERENCE;
+    IfFailRet(JsGetPropertyIdFromName(L"console", &consolePropertyId));
+
+    JsValueRef consoleObject = JS_INVALID_REFERENCE;
+    IfFailRet(JsGetProperty(globalObject, consolePropertyId, &consoleObject));
+
+    JsValueRef undefinedValue = JS_INVALID_REFERENCE;
+    IfFailRet(JsGetUndefinedValue(&undefinedValue));
+
+    if (consoleObject == JS_INVALID_REFERENCE || consoleObject == undefinedValue)
+    {
+        return JsNoError;
+    }
+
+    JsValueRef debuggerConsoleObject = JS_INVALID_REFERENCE;
+    IfFailRet(handler->GetConsoleObject(&debuggerConsoleObject));
+
+    std::string script = "";
+
+    script = script + "function patchConsoleObject$$1(global, hostConsole, debugConsole) {\n";
```

Comment:
I would prefer another name for "hostConsole". I think it could just be "console" but if you want to distinguish it could be "existingConsole", "consoleToPath", etc. Not sure what the best name is...